src/Middleware/DispatchMiddleware.php

```diff
@@ -40,6 +78,15 @@ public function process(ServerRequestInterface $request, DelegateInterface $dele
 
         $middleware = $routeResult->getMatchedMiddleware();
 
+        if (! $middleware instanceof ServerMiddlewareInterface) {
+            $middleware = $this->prepareMiddleware(
+                $middleware,
+                $this->router,
+                $this->responsePrototype,
+                $this->container
+            );
```

Comment:
By adding the `prepareMiddleware` method here, I guess it can be removed now from [Application::route](https://github.com/weierophinney/zend-expressive/blob/a14315c04713d5463920c9e0853857f5f46de829/src/Application.php#L314-L319)?